fix outstanding issues after #1315

hmmr opened the pull request:
Fix outstanding dialyzer issues after https://github.com/basho/riak_kv/pull/1315.

Depends on https://github.com/basho/riak_ql/pull/80. A r_t module checking some window functions is https://github.com/basho/riak_test/pull/966.

Andalso, fix value returned by `riak_kv_pb_timeseries:get_column_types/2`.

hmmr committed:
correctly receive results from qry:submit, now returning a 3-tuple

hmmr committed:
fix pb_timeseries:get_column_types

hmmr committed:
attempts to deal with "no local return" in select_column_clause_folder/3

hmmr committed:
pedantic tweak of spec of run_select/2

hmmr committed:
unmuffle dialyzer

hmmr committed:
put types in order to stub "no local return" in select_column_clause_folder

hmmr committed:
don't flatten list of functions

gordonguthrie commented on src/riak_kv_qry_compiler.erl:
```diff
@@ -156,7 +156,7 @@ select_column_clause_folder(DDL, ColX, {TypeSet1, SelClause1}) ->
         clause           = RunFn2,
         is_valid         = IsValid2,
-        finalisers       = lists:flatten(Finalisers2)},
+        finalisers       = Finalisers2},
```
This will (probably) fail with a select \* clause

gordonguthrie commented on src/riak_kv_qry_compiler.erl:
```diff
@@ -210,24 +211,23 @@ compile_select_col(DDL, {{window_agg_fn, FnName}, [FnArg1]}) ->
                         clause           = fun(Row, _) -> Fn(Row) end,
                         is_valid         = IsValid,
-                        finaliser        = Finalisers }.
+                        finaliser        = fun(X) -> X end }.
```
Not sure this is right either?

hmmr committed:
Revert "don't flatten list of functions"

This reverts commit 9ec76d092f65e888e5a7fd17afbafb90ca40478f.

Sayeth Gordon: "because a select * clause will expand to a list of
finalise functions, one for each column and you will get a list of
lists in there"

hmmr committed:
for stateless functions, set #single_set_column.finaliser to []

hmmr committed:
recognise signatures of functions accepting '*'

hmmr committed:
Merge branch 'end-to-end/timeseries' into fix/az/refactor-dialyzer-coverup

# Conflicts:
#	src/riak_kv_qry_compiler.erl

hazen commented:
@hmmr LOL. You are doing exactly what I was doing.  I was going to test your branch, but needed to merge in `e2e/ts`.  Since it was only 4:00am in the Ukraine, I thought you **might** be asleep.

hmmr committed:
tiny fix of comment indent after emacs

hmmr committed:
regression fix in riak_kv_pb_timeseries:get_column_types/2

This fixes the following badness:

 [error] <0.1925.0> gen_fsm <0.1925.0> in state connected terminated
  with reason: {error,function_clause,[{'riak_ql_table_GeoCheckin$1',
  get_field_type,[<<"geohash">>],[{file,[]},{line,7}]},
  {riak_kv_pb_timeseries,'-get_column_types/2-lc$^0/1-0-',2,

andytill commented:
+1  777212a
